src/main/resources/db/migration/V17__add_collections_enclosure_table.sql

```diff
@@ -0,0 +1,18 @@
+CREATE TABLE collections_enclosure (
+    id UUID PRIMARY KEY DEFAULT public.gen_random_uuid(),
+    collections_id UUID REFERENCES collections_details(id) NOT NULL,
+    title VARCHAR(255) NOT NULL,
+    href VARCHAR(255) NOT NULL,
+    type VARCHAR(50) NOT NULL,
+    size BIGINT NOT NULL,
+    CONSTRAINT unique_enclosure UNIQUE (collections_id, title)
+);
+
+INSERT INTO collections_enclosure (id, collections_id, title, href, type, size)
+SELECT id, stac_collections_id, title, href, type, size
+FROM stac_collections_assets;
```

Comment:
You can add a `truncate` statement to the `stac_collections_assets` table since all the data will now be moved to a new table and all existing rows won't have any relation with the new data that will added in the future.